pkg/solana/txm/txm_internal_test.go

```diff
@@ -1186,3 +1209,432 @@ func addSigAndLimitToTx(t *testing.T, keystore SimpleKeystore, pubkey solana.Pub
 	require.NoError(t, fees.SetComputeUnitLimit(&txCopy, limit))
 	return &txCopy
 }
+
+func TestTxm_ExpirationRebroadcast(t *testing.T) {
```

Comment:
Hey! We've found that testing this scenario without mocking is challenging because we don't have a way to force a transaction to expire in a live validator.

The only solution we've identified involves modifying some methods to simulate this behavior for the tests. However, this approach is not ideal, as it would require adding additional flows solely to make the test pass and to trigger a rebroadcast in the live validator tests.

Do you have any specific ideas on how to achieve this that we might be missing?

Reply:
you could potentially introduce some sort of proxy that takes the transaction and returns a transaction signature, but doesn't allow any more transactions through until it knows the first transaction blockhash has expired

Reply:
Would it be okay to address this in a separate ticket under the [Testing](https://smartcontract-it.atlassian.net/browse/NONEVM-698) track? So we are able to get this one in before the freeze and finish the re-org work that depends on this PR

Reply:
why does the freeze impact the timeline? afaik that's just for production environments, not development?

also this PR touches significant portions of the most sensitive part of a solana integration, test coverage imo is a must

Reply:
just clarifying a few things:
* there is a production code freeze, but that shouldn't impact development here
* i'd like to see an attempt at a proper integration tests, but it can be time-boxed to a day or two (if it still doesn't work, then a follow-up ticket is fine)

Reply:
Thanks for clarifying, thought the freeze involved not merging new code in `chainlink repo`, even if it was development 😅. I'll start working on these integration tests